### ISSUE TITLE
v4.0.0-beta.7 - when /profile request returns 401 (Unauthhorized) should be handle somehow

### ISSUE THREAD

**Issue by sleitor** [auth0/nextjs-auth0#1817]
### Checklist

- [X] The issue can be reproduced in the [nextjs-auth0 sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Heed some hook, ability, or automatically catch 401 error on /profile (that request by use-user.js on 15th line) to force logout from the app.

How we can catch it and redirect to /login or /logout page?

### Reproduction

1) login on auth0
2) in the developer tab remove the cookie 
3) focus off/on on the page or just wait couple time
4) after get atomatically on middleware will get a 401 for profile

### nextjs-auth0 version

v4.0.0-beta.7

### Next.js version

15.0.3

### Node.js version

v20.18.0

**Comment by guabu** [comment]
Hey @sleitor 👋 When the call to `/profile` returns the 401 the user will no longer be defined. You should be able to check the existence of the user (it will be `null` if they're not authenticated) and act on that:

```tsx
"use client";
 
import { useUser } from "@auth0/nextjs-auth0";
 
export default function Profile() {
  const { user, isLoading } = useUser();
  
  if (!user) {
    // handle the logic you want here
  }
 
  return (
    // ...
  );
}
```

**Comment by sleitor** [comment]
Thank you @guabu! Appreciate for fast answer.

But Unfrotunatelly this not work as expected. 
If user authorized first result there is `undefined` and after couple seconds returns the user profile.

But bigger problem happen after.

when you remove the __session from cookies then we get 401 for /profile, but useUser continue returning the user profile and no issues.

<img width="383" alt="image" src="https://github.com/user-attachments/assets/627fdcb2-da4e-4e8f-b44b-5e0e59915b7a">

**Comment by guabu** [comment]
Hey @sleitor 👋 Thanks for the follow-up!

We've got a fix that should address this in the next release as well!

**Comment by nandan-bhat** [comment]
Closing this ticket as this issue has been fixed `4.0.0-beta.9`
